### ISSUE TITLE
Can not produce the transfer learning result 82.8% on ImageNet.  Ask for fine-tuning code and hyper-parameters for transfer learning on ImageNet.

### ISSUE THREAD

**Issue by KimTakLim1996** [facebookresearch/dino#31]
Thanks for your wonderful work!

I notice the amazing transfer learning result in Appendix A.
Can I direct use the code and parameters configuration in [DEIT](https://github.com/facebookresearch/deit) to do fine-tuning and then get the 82.8 top-1 performance on ImageNet?
Is it necessary to change the hyper-parameters of fine-tuning or change the way of data aug? I noticed that DINO and DEIT use different data aug methods.

I use the code and hyper-parameters in [DEIT](https://github.com/facebookresearch/deit) to fine-tuning DINO-base/16, but only got 81.34% top-1 acc on ImageNet. Which is even worse than the from scratch setting 81.8% top-1 .

**Comment by mathildecaron31** [comment]
Hi @KimTakLim1996 

Thanks for your interest and your kind words.

Yes I've used the DeiT codebase for the finetuning. The data augmentation at finetuning time is exactly the same as in default DeiT training parameters. The base learning rate is `0.0005`, as I train with batch size of 1024 (16 GPUs) this makes an effective learning rate of `0.001`.

There is one difference in my implementation compared to DeiT which is that I initialize the linear classification layer (the only one to start from random weights) with smaller values (basically I multiply the weights by 0.001). I have not evaluated finetuning without this trick so I cannot tell you if this is what explains the difference.

I join my [training logs](https://dl.fbaipublicfiles.com/dino/dino_vitbase16_finetune/dino_vitbase16_finetune_log.txt) and the [final checkpoint](https://dl.fbaipublicfiles.com/dino/dino_vitbase16_finetune/dino_vitbase16_finetune.pth).

Hope that helps.

**Comment by KimTakLim1996** [comment]
Hi @mathildecaron31 , 
I share my [training logs](https://gist.github.com/KimTakLim1996/e5c67783b4a24f2ef5315edf2c4e4f04) for my finetuning experiments:

This [log](https://gist.github.com/KimTakLim1996/e5c67783b4a24f2ef5315edf2c4e4f04#file-deit_g8bsz128-from-scratch) is training a deit-base-patch16 from scratch, which got 81.8 performance.

[This log](https://gist.github.com/KimTakLim1996/e5c67783b4a24f2ef5315edf2c4e4f04#file-deit_g8bsz128-from-dino-base_patch16) use the same code and config to finetune the DINO-base-16 (backbone-only), only got 81.4.

And [this log](https://gist.github.com/KimTakLim1996/e5c67783b4a24f2ef5315edf2c4e4f04#file-deit_g8bsz128-from-dino-base_patch16-class-head-weight-init-mul-0-001) multiply the class head's weights by 0.001. I observe the first two epoch got much higher performance than last setting (40% + vs 0~2%), but got the close performance 81.4 finally.

Can you help me to reproduce the experimental results? The code I used has not changed and can successfully reproduce the performance 81.8 of deit. However, there is a big gap in Dino's transfer learning setting.

I run my experiment on 8x32G v100, which support me to set 128 batch size per GPU.
My training command is:

```bash
python -m torch.distributed.launch --nproc_per_node=8 --use_env main.py \
        --data-path /data/ILSVRC/Data/CLS-LOC/train \
        --output_dir /data/output_for_dino_finetuning \
        --model deit_base_patch16_224 --lr 5e-4 --init_scale 0.001 --batch-size 128 \
        --finetune /data/ckpts/dino_vitbase16_pretrain.pth --no-model-ema
```

**Comment by KimTakLim1996** [comment]
I found the issue from https://github.com/facebookresearch/deit/blob/main/main.py#L379, the deit code will disable dropout during fine-tuning by default. I can get a close number to 82.8 when I enable dropout during fine-tuning.
@mathildecaron31 Thanks for your help!
